src/lua/uri.lua

```diff
@@ -218,6 +253,50 @@ local function values(...)
     return {...}
 end
 
+local char_to_hex = function(c)
+    return string.format("%%%02X", string.byte(c))
+end
+
+-- Encodes a string into its escaped hexadecimal representation.
+local function str_escape(buf, opts)
+    if type(buf) ~= "string" then
+        error("Usage: str_escape(string, [opts])")
+    end
+    opts = opts or RFC3986
+    opts.unreserved = opts.unreserved or RFC3986.unreserved
+    opts.plus = opts.plus or RFC3986.plus
+
+    local url = buf
```

Comment:
We have nothing to do with URLs in this function. It is confusing to name a source data to encode into an URI as `url`.

Reply:
The same about the testing code. It is important, as for me. I wouldn't pass url/uri to a function that encodes a query parameter value, this action has no much sense (if we don't want to pass it as data within a query parameter, of course).